flw/src/main/java/org/motechproject/nms/flw/domain/WhitelistEntry.java

```diff
@@ -0,0 +1,72 @@
+package org.motechproject.nms.flw.domain;
+
+import org.motechproject.mds.annotations.Entity;
+import org.motechproject.mds.annotations.Field;
+import org.motechproject.nms.location.domain.State;
+
+import javax.jdo.annotations.Column;
+import javax.jdo.annotations.Unique;
+import javax.validation.constraints.NotNull;
+
+@Entity(tableName = "whitelist_entries")
+@Unique(name = "UNIQUE_STATE_CONTACT_NUMBER_COMPOSITE_IDX", members = { "state", "contactNumber" })
+public class WhitelistEntry {
+    public static final int FIELD_SIZE_10 = 10;
+
+    @Field
+    @NotNull
+    private State state;
```

Comment:
I'm not sure I follow, but I think you have a good point.  We never really
load the whitelist entry, just check it's existence.  When users upload
data to populate it they will use the state code, not our state id.  (we
will need to validate on upload that we have a state for that state code
(I'll see if there is a jdo annotation that can set that foreign key))

On Thu, Apr 23, 2015 at 11:22 AM, kosh <notifications@github.com> wrote:

> In flw/src/main/java/org/motechproject/nms/flw/domain/WhitelistEntry.java
> <https://github.com/koshalt/mim/pull/74#discussion_r28989113>:
>
> > +import org.motechproject.mds.annotations.Entity;
> > +import org.motechproject.mds.annotations.Field;
> > +import org.motechproject.nms.location.domain.State;
> > +
> > +import javax.jdo.annotations.Column;
> > +import javax.jdo.annotations.Unique;
> > +import javax.validation.constraints.NotNull;
> > +
> > +@Entity(tableName = "whitelist_entries")
> > +@Unique(name = "UNIQUE_STATE_CONTACT_NUMBER_COMPOSITE_IDX", members = { "state", "contactNumber" })
> > +public class WhitelistEntry {
> > +    public static final int FIELD_SIZE_10 = 10;
> > +
> > +    @Field
> > +    @NotNull
> > +    private State state;
>
> Would the whitelist entry be a 'full' state or just a statecode and a
> contact number can then be looked up?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/koshalt/mim/pull/74/files#r28989113>.
>